### ISSUE TITLE
version 1.4.0 significantly slows down site generation on macOS

### ISSUE THREAD

**Issue by offbyone** [pelican-plugins/photos#70]
I was digging in to find out why my site generation had slowed down so much after updating dependencies, and I narrowed it down to this plugin. When generating a site from ~640 articles, most of which don't have galleries or images, my site with 1.3.0 pinned takes 15 seconds, and when I upgrade to 1.4.0 that time jumps to over 700 seconds.

I'm not sure why :D Happy to help troubleshoot!

**Comment by phibos** [comment]
I had a look at you issue. I have tried to setup your blog but there where some errors. Maybe I'm missing something. Some node modules or so.

So I have copied all the files from your "content/old" directory into the example articles folder and tried to build the blog.

Everything is done with the current "main" branch of the photos plugin which should be the same code as version 1.4.0.
Python 3.10.8 and Linux because I don't have a Mac.

With pelican 4.7.1
```
$ pip freeze
blinker==1.4
colorama==0.4.4
commonmark==0.9.1
docutils==0.18.1
feedgenerator==2.0.0
Jinja2==3.0.3
Markdown==3.3.6
MarkupSafe==2.0.1
pelican==4.7.1
piexif==1.1.3
Pillow==8.4.0
Pygments==2.11.1
python-dateutil==2.8.2
pytz==2021.3
rich==10.16.2
six==1.16.0
Unidecode==1.3.2
```

```
"pelican" -lr "/home/user/git/github/pelican-plugins/photos/examples/content" -o "/home/user/git/github/pelican-plugins/photos/examples/output" -s "/home/user/git/github/pelican-plugins/photos/examples/pelicanconf.py" 
  --- AutoReload Mode: Monitoring `content`, `theme` and `settings` for changes. ---
Serving site at: http://127.0.0.1:8000 - Tap CTRL-C to stop
[21:14:42] WARNING  Watched path does not exist: /home/user/git/github/pelican-plugins/photos/examples/content/images                                                                                                               log.py:91

-> Modified: settings, content, theme. re-generating...
Done: Processed 24 articles, 603 drafts, 0 hidden articles, 6 pages, 0 hidden pages and 1 draft page in 12.31 seconds.

-> Modified: content. re-generating...
Done: Processed 24 articles, 603 drafts, 0 hidden articles, 6 pages, 0 hidden pages and 1 draft page in 11.27 seconds.
````

With pelican 4.8.0

```
$ pip freeze
blinker==1.5
commonmark==0.9.1
docutils==0.19
feedgenerator==2.0.0
Jinja2==3.1.2
MarkupSafe==2.1.1
pelican==4.8.0
piexif==1.1.3
Pillow==9.3.0
Pygments==2.13.0
python-dateutil==2.8.2
pytz==2022.6
rich==12.6.0
six==1.16.0
```

````
"pelican" -lr "/home/user/git/github/pelican-plugins/photos/examples/content" -o "/home/user/git/github/pelican-plugins/photos/examples/output" -s "/home/user/git/github/pelican-plugins/photos/examples/pelicanconf.py" 
  --- AutoReload Mode: Monitoring `content`, `theme` and `settings` for changes. ---
Serving site at: http://127.0.0.1:8000 - Tap CTRL-C to stop
[21:23:16] WARNING  Watched path does not exist: /home/user/git/github/pelican-plugins/photos/examples/content/images                                                                                                               log.py:91

-> Modified: settings, content, theme. re-generating...
Done: Processed 19 articles, 602 drafts, 0 hidden articles, 6 pages, 0 hidden pages and 1 draft page in 10.46 seconds.


-> Modified: settings. re-generating...
Done: Processed 19 articles, 602 drafts, 0 hidden articles, 6 pages, 0 hidden pages and 1 draft page in 11.16 seconds.
```

Generating the blog with over 600 articles in 11 seconds seams good to me. But I'm unable to figure out why it is so slow on your Mac. Can you run pelican with a Python profiler and filter for the photos plugin so we can identify the function which requires most of the processing time.

**Comment by offbyone** [comment]
I would absolutely love to do that, but I haven't managed to figure out how to get pelican to run under a profiler. Every profiler that I've tried simply profiles the main process, but pelican seems to farm out to multiprocessing quite a bit. If you have any suggestions as to how I can get it to go under one I would be happy to do so (that was what I was trying to do before I managed to triangulate on this plug-in in the first place 😁)

**Comment by phibos** [comment]
The easiest way is to run pelican in Debug mode by adding the ```--debug``` flag. This will output something like the following on the first run.
```
           DEBUG    photos: Open file content/galleries/snoqualmie-2021/IMG_8999.png                                                                                                                                            photos.py:1352
           DEBUG    STREAM b'iCCP' 41 26924                                                                                                                                                                              PngImagePlugin.py:201
           DEBUG    iCCP profile name b'ICC Profile'                                                                                                                                                                     PngImagePlugin.py:406
           DEBUG    Compression method 0                                                                                                                                                                                 PngImagePlugin.py:407
           DEBUG    STREAM b'eXIf' 26977 2412                                                                                                                                                                            PngImagePlugin.py:201
           DEBUG    STREAM b'pHYs' 29401 9                                                                                                                                                                               PngImagePlugin.py:201
           DEBUG    STREAM b'iTXt' 29422 4991                                                                                                                                                                            PngImagePlugin.py:201
           DEBUG    STREAM b'IDAT' 34425 16384                                                                                                                                                                           PngImagePlugin.py:201
           DEBUG    tag: DateTime (306) - type: string (2) Tag Location: 94 - Data Location: 202 - value: b'2021:09:10 15:27:45\x00'                                                                                    TiffImagePlugin.py:842
           DEBUG    tag: GPSInfoIFD (34853) - type: long (4) - value: b'\x00\x00\x08z'                                                                                                                                  TiffImagePlugin.py:842
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg): 0=exif.rotate                                                                                photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg): 1=exif.manipulate                                                                            photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg): 2=main.remove_alpha                                                                          photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg): 3=main.resize                                                                                photos.py:1047
[21:05:17] DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg): 4=main.watermark                                                                             photos.py:1047
           INFO     photos: make photo(PID: 1263391) content/galleries/snoqualmie-2021/IMG_8999.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg                       photos.py:1003
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg): 0=exif.rotate                                                                               photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg): 1=exif.manipulate                                                                           photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg): 2=main.remove_alpha                                                                         photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg): 3=main.resize                                                                               photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg): 4=main.watermark                                                                            photos.py:1047
           INFO     photos: make photo(PID: 1263391) content/galleries/snoqualmie-2021/IMG_9005.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg                        photos.py:1003
           DEBUG    photos: Open file content/galleries/snoqualmie-2021/IMG_9005.png                                                                                                                                            photos.py:1352
           DEBUG    STREAM b'eXIf' 54 2444                                                                                                                                                                               PngImagePlugin.py:201
           DEBUG    STREAM b'pHYs' 2510 9                                                                                                                                                                                PngImagePlugin.py:201
           DEBUG    STREAM b'iTXt' 2531 5055                                                                                                                                                                             PngImagePlugin.py:201
           DEBUG    STREAM b'IDAT' 7598 16384                                                                                                                                                                            PngImagePlugin.py:201
           DEBUG    tag: DateTime (306) - type: string (2) Tag Location: 94 - Data Location: 202 - value: b'2021:09:10 15:29:25\x00'                                                                                    TiffImagePlugin.py:842
           DEBUG    tag: GPSInfoIFD (34853) - type: long (4) - value: b'\x00\x00\x08\x9a'                                                                                                                               TiffImagePlugin.py:842
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg): 0=exif.rotate                                                                                photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg): 1=exif.manipulate                                                                            photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg): 2=main.remove_alpha                                                                          photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg): 3=main.resize                                                                                photos.py:1047
[21:05:18] DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg): 4=main.watermark                                                                             photos.py:1047
           INFO     photos: make photo(PID: 1263391) content/galleries/snoqualmie-2021/IMG_9005.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg                       photos.py:1003
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg): 0=exif.rotate                                                                               photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg): 1=exif.manipulate                                                                           photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg): 2=main.remove_alpha                                                                         photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg): 3=main.resize                                                                               photos.py:1047
           DEBUG    Processing(/home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg): 4=main.watermark                                                                            photos.py:1047
           INFO     photos: make photo(PID: 1263391) content/galleries/snoqualmie-2021/IMG_9006.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9006.jpg                        photos.py:1003
```

On the next runs this should be much faster and should show something like in the example below.

```
           INFO     photos: 16 images in resize queue                                                                                                                                                                           photos.py:1564
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8996.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8996.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.3813913 1670961914.6116214                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8996.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8996t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.3813913 1670961914.9716213                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8998.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8998.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.4047246 1670961916.1716213                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8998.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8998t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.4047246 1670961916.6516213                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8999.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.4380581 1670961917.118288                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_8999.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_8999t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.4380581 1670961917.4982882                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9005.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.4613914 1670961918.048288                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9005.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9005t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.4613914 1670961918.5249548                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9006.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9006.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.4847248 1670961919.144955                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9006.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9006t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.4847248 1670961919.554955                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9009.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9009.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.5047247 1670961920.184955                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9009.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9009t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.5047247 1670961920.594955                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9011.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9011.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.5247247 1670961921.234955                                                                                                                                                         photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9011.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9011t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.5247247 1670961921.6682885                                                                                                                                                        photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9012.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9012.jpg                        photos.py:1003
           DEBUG    Skipping orig: 1670874609.558058 1670961922.224955                                                                                                                                                          photos.py:1011
           INFO     photos: make photo(PID: 1266662) content/galleries/snoqualmie-2021/IMG_9012.png -> /home/user/git/github/pelican-plugins/photos/examples/output/photos/snoqualmie-2021/img_9012t.jpg                       photos.py:1003
           DEBUG    Skipping orig: 1670874609.558058 1670961922.7316217                                                                                                                                                         photos.py:1011
```

You will see a timestamp if the time elapsed is at least 1 second. This should give some hints where most of the time has been spent.

If you want to dig deeper into the code you can use Pythons cProfile module. You can find an example how to use it in the PR #71.

**Comment by offbyone** [comment]
The part that is slow is the other posts:

```
[23:43:29] DEBUG    Read file old/general-thoughts/upgrades.rst -> Article                      readers.py:547
[23:43:30] DEBUG    Read file old/internet/oogling-google.rst -> Article                        readers.py:547
[23:43:31] DEBUG    Read file old/events/gorged.rst -> Article                                  readers.py:547
[23:43:32] DEBUG    Read file old/events/if-hockeylocalteamwins-hornhonkstatelike_mad.rst ->    readers.py:547
                    Article
```

... for example, you can see each of the earlier articles are taking a second to render

That's the case for every single article in my old posts.

**Comment by phibos** [comment]
I think I have found the reason for the slow down. The plugin creates a process pool for every page/article even if there are no images to process. I have found some information pointing out that the fork() in Python to create a new process is very 'expensive'(time consuming). I have changed the processing to use a global pool and reuse it. If you are interested have a look at PR #72

**Comment by offbyone** [comment]
_Massively faster_, thanks!

One problem, though: 

```
/photos.py", line 1598, in process_image_process_wrapper
    return image.process()
  File "/Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/plugins/photos/photos.py", line 1063, in process
    if pelican_settings["PHOTO_EXIF_KEEP"] and "exif" in image.info:
KeyError: 'PHOTO_EXIF_KEEP'
           CRITICAL TypeError: cannot convert dictionary update sequence element #0 to a       __init__.py:566
                    sequence
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/__init__ │
│ .py:562 in main                                                                                  │
│                                                                                                  │
│   559 │   │   │   watcher = FileSystemWatcher(args.settings, Readers, settings)                  │
│   560 │   │   │   watcher.check()                                                                │
│   561 │   │   │   with console.status("Generating..."):                                          │
│ ❱ 562 │   │   │   │   pelican.run()                                                              │
│   563 │   except KeyboardInterrupt:                                                              │
│   564 │   │   logger.warning('Keyboard interrupt received. Exiting.')                            │
│   565 │   except Exception as e:                                                                 │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/__init__ │
│ .py:121 in run                                                                                   │
│                                                                                                  │
│   118 │   │   │   if hasattr(p, 'refresh_metadata_intersite_links'):                             │
│   119 │   │   │   │   p.refresh_metadata_intersite_links()                                       │
│   120 │   │                                                                                      │
│ ❱ 121 │   │   signals.all_generators_finalized.send(generators)                                  │
│   122 │   │                                                                                      │
│   123 │   │   writer = self._get_writer()                                                        │
│   124                                                                                            │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/blinker/base.py: │
│ 263 in send                                                                                      │
│                                                                                                  │
│   260 │   │   │   │   │   │   │   '%s given' % len(sender))                                      │
│   261 │   │   else:                                                                              │
│   262 │   │   │   sender = sender[0]                                                             │
│ ❱ 263 │   │   return [(receiver, receiver(sender, **kwargs))                                     │
│   264 │   │   │   │   for receiver in self.receivers_for(sender)]                                │
│   265 │                                                                                          │
│   266 │   def has_receivers_for(self, sender):                                                   │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/blinker/base.py: │
│ 263 in <listcomp>                                                                                │
│                                                                                                  │
│   260 │   │   │   │   │   │   │   '%s given' % len(sender))                                      │
│   261 │   │   else:                                                                              │
│   262 │   │   │   sender = sender[0]                                                             │
│ ❱ 263 │   │   return [(receiver, receiver(sender, **kwargs))                                     │
│   264 │   │   │   │   for receiver in self.receivers_for(sender)]                                │
│   265 │                                                                                          │
│   266 │   def has_receivers_for(self, sender):                                                   │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/plugins/ │
│ photos/photos.py:192 in measure_time_wrapper                                                     │
│                                                                                                  │
│    189 │   @wraps(func)                                                                          │
│    190 │   def measure_time_wrapper(*args, **kwargs):                                            │
│    191 │   │   if not pelican_settings["PHOTO_PROFILING_ENABLED"]:                               │
│ ❱  192 │   │   │   return func(*args, **kwargs)                                                  │
│    193 │   │                                                                                     │
│    194 │   │   global g_profiling_call_level                                                     │
│    195                                                                                           │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/plugins/ │
│ photos/photos.py:2122 in handle_signal_all_generators_finalized                                  │
│                                                                                                  │
│   2119 │   │   │   │   detect_meta_galleries(page)                                               │
│   2120 │   │   │   │   page.photo_global_images = global_images                                  │
│   2121 │                                                                                         │
│ ❱ 2122 │   process_image_queue()                                                                 │
│   2123                                                                                           │
│   2124                                                                                           │
│   2125 def register():                                                                           │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/plugins/ │
│ photos/photos.py:192 in measure_time_wrapper                                                     │
│                                                                                                  │
│    189 │   @wraps(func)                                                                          │
│    190 │   def measure_time_wrapper(*args, **kwargs):                                            │
│    191 │   │   if not pelican_settings["PHOTO_PROFILING_ENABLED"]:                               │
│ ❱  192 │   │   │   return func(*args, **kwargs)                                                  │
│    193 │   │                                                                                     │
│    194 │   │   global g_profiling_call_level                                                     │
│    195                                                                                           │
│                                                                                                  │
│ /Users/offby1/projects/ideas/.direnv/python-3.10.2/lib/python3.10/site-packages/pelican/plugins/ │
│ photos/photos.py:1621 in process_image_queue                                                     │
│                                                                                                  │
│   1618 │   │   │   k, info = image.process()                                                     │
│   1619 │   │   │   results[k] = info                                                             │
│   1620 │   else:                                                                                 │
│ ❱ 1621 │   │   results = dict(                                                                   │
│   1622 │   │   │   g_process_pool.imap_unordered(process_image_process_wrapper, image_queue)     │
│   1623 │   │   )                                                                                 │
│   1624                                                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: cannot convert dictionary update sequence element #0 to a sequence
```

I have `PHOTO_EXIF_KEEP` set to `True` and `image.info` is apparently not the same type as it used to be.

**Comment by phibos** [comment]
I have fixed some minor issues in PR #73 and I think I have found the issue with multiprocessing and the error above. I hope to fix it in the next few minutes.

**Comment by offbyone** [comment]
Brilliant, thank you! I've confirmed with the tip of your mainline branch. All good now. I'll switch to 1.5.0 when you release it.
